zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -67,18 +95,39 @@ impl Chain {
 
     /// Fork a chain at the block with the given hash, if it is part of this
     /// chain.
-    pub fn fork(&self, fork_tip: block::Hash) -> Option<Self> {
+    ///
+    /// `finalized_tip_history_tree`: the history tree for the finalized tip
+    /// from which the tree of the fork will be computed.
+    pub fn fork(
+        &self,
+        fork_tip: block::Hash,
+        finalized_tip_history_tree: &HistoryTree,
+    ) -> Result<Option<Self>, HistoryTreeError> {
```

Comment:
API usability: let's use a more general error type `ValidateContextError`
```suggestion
    ) -> Result<Option<Self>, ValidateContextError> {
```